src/VitessceGrid.js

```diff
@@ -41,101 +40,100 @@ function shallowEqual(objA, objB) {
   return true;
 }
 
-
-export default class VitessceGrid extends React.Component {
-  constructor(props) {
-    super(props);
-    this.state = { readyComponentKeys: new Set() }; // eslint-disable-line react/no-unused-state
-    // This is being used and I'm not sure why there's a linting problem.
-  }
-
-  shouldComponentUpdate(nextProps) {
-    // We are ignoring state right now and only considering props.
-    // State is updated as components trigger onReady...
-    // but those events should not cause a re-render.
-    // React docs warn:
-    // > Do not rely on it to “prevent” a rendering, as this can lead to bugs.
-    // so there is probably a better approach.
-    return !shallowEqual(this.props, nextProps);
-  }
-
-  render() {
-    const {
-      layout, getComponent, padding, margin, draggableHandle,
-      reactGridLayoutProps, onAllReady, rowHeight,
-    } = this.props;
-
-    const ResponsiveGridLayout = WidthProvider(Responsive);
-
-    const {
-      cols, layouts, breakpoints, components,
-    } = resolveLayout(layout);
-
-    // Inline CSS is generally avoided, but this saves the end-user a little work,
-    // and prevents class names from getting out of sync.
-    const style = (
-      <style>
-        {`
+const VitessceGridComponent = ({
+  k, v, Component, onReady, removeGridComponent,
+}) => (
+  <div key={k}>
+    <Component
+      {... v.props}
+      clearComponent={removeGridComponent}
+      onReady={onReady}
+    />
+  </div>
+);
+
+const VitessceGrid = (props) => {
+  const {
+    layout, getComponent, padding, margin, draggableHandle,
+    reactGridLayoutProps, onAllReady, rowHeight,
+  } = props;
+  const ResponsiveGridLayout = WidthProvider(Responsive);
+  const {
+    cols, layouts, breakpoints, components,
+  } = resolveLayout(layout);
```

Comment:
Vitessce isn't using responsive any longer... but I'm ok leaving this a little more flexible for now, in case that decision is revisited.

Reply:
I'd be for leaving this right now, as I doubt it adds much to the bundle and allows it to be more flexible in the future.